src/rt/aaA.d

```diff
@@ -672,25 +672,28 @@ extern (C) int _aaEqual(in TypeInfo tiRaw, in AA aa1, in AA aa2)
 extern (C) hash_t _aaGetHash(in AA* aa, in TypeInfo tiRaw) nothrow
 {
     if (aa.empty)
-        return 0;
+        return hashOf(0);
```

Comment:
Suggestion: revert this to `return 0` and change `core.internal.hash` to match.

Reply:
`core.internal.hash.hashOf` may use seed value.
The current function looks like
```
size_t hashOf(T)(T aa, size_t seed = 0) if (!is(T == enum) && __traits(isAssociativeArray, T))
{
    if (!aa.length) return hashOf(0, seed);
}
```

If I rewrite it as...
```
size_t hashOf(T)(T aa, size_t seed = 0) if (!is(T == enum) && __traits(isAssociativeArray, T))
{
    if (!aa.length) return seed;
}
```
... we will disable way to chain several hashes:
```
a = hashOf(cast(int[int])null, hashOf(cast(int[int])null));
b = hashOf(cast(int[int])null, hashOf(cast(int[int])null, hashOf(cast(int[int])null)));
// a == b but a should differs b
```
as result:
hashes of two arrays with different count of null AA will be same:
```
auto arr = [int[int].init, int[int].init];
auto arr2 = [int[int].init, int[int].init, int[int].init];
assert(hashOf(arr) != hashOf(arr2)); // will be failed
```